components/prism-vala.js

```diff
@@ -0,0 +1,68 @@
+Prism.languages.vala = Prism.languages.extend('clike', {
+	// Classes copied from prism-csharp
+	'class-name': [
+		{
+			// (Foo bar, Bar baz)
+			pattern: /\b[A-Z]\w*(?:\.\w+)*\b(?=\s+\w+)/,
```

Comment:
It's quite common for types to be nullable. E.g.: `Color? c = getColor();`
Also pointers? (`Foo* f_ptr = f;`)

Reply:
I copied the classes from the C# component, so I'm guessing pointers should be added there as well.

Reply:
That's not necessary. Pointers have a few constraints in C#, so that shouldn't be necessary (as far as I know).